### PR TITLE
MOD-5869: Support mariner2 via github actions

### PR DESCRIPTION
**Pull request by nafraf**
Support Mariner2 Linux using github actions.

### CODE REVIEW

File: .install/get_redis.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+curdir="$PWD"
+cd /tmp/
+# Download redis source from github archive and extract it.
+wget https://github.com/redis/redis/archive/${REDIS_VERSION}.tar.gz
+tar -xvzf ${REDIS_VERSION}.tar.gz && cd ./redis-${REDIS_VERSION}
+
+#patch
+sed -ri 's/(createEnum.*enable-protected-configs.*PROTECTED_ACTION_ALLOWED)_NO/\1_YES/g' ./src/config.c
+sed -ri 's/(createEnum.*enable-debug-command.*PROTECTED_ACTION_ALLOWED)_NO/\1_YES/g' ./src/config.c
+sed -ri 's/(createEnum.*enable-module-command.*PROTECTED_ACTION_ALLOWED)_NO/\1_YES/g' ./src/config.c
```
Comment (alonre24) [inline .install/get_redis.sh:14]: Is this needed? Don't we use RLTest for that?

File: .github/workflows/mariner2.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        run: |
+          make build
+      - name: test
+        run: |
+          make test
+      - name: pack module
+        run: |
+          make pack BRANCH=${{ github.ref_name }}
+      - name: get aws
+        # TODO: New script .install/get_aws.sh?
```
Comment (alonre24) [inline .github/workflows/mariner2.yml:38]: I just added the `aws` install to the dependencies script

File: .github/workflows/mariner2.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          bash .install/get_redis.sh
+      - name: build
+        run: |
+          make build
+      - name: test
+        run: |
+          make test
+      - name: pack module
+        run: |
+          make pack BRANCH=${{ github.ref_name }}
```
Comment (alonre24) [inline .github/workflows/mariner2.yml:36]: why did you add `BRANCH=${{ github.ref_name }}`?

Comment (nafraf) [reply .github/workflows/mariner2.yml:36]: I was getting this error of permissions when pack.sh tries to get the branch name, and altough I tried to solve it in the checkout action it did not work.

```
fatal: detected dubious ownership in repository at '/__w/RedisJSON/RedisJSON'
To add an exception for this directory, call:

	git config --global --add safe.directory /__w/RedisJSON/RedisJSON

In /__w/RedisJSON/RedisJSON/sbin/pack.sh:
239      #----------------------------------------------------------------------------------------------
240      
241      if [[ -z $BRANCH ]]; then
242  >>> 	BRANCH=$(git rev-parse --abbrev-ref HEAD)
243      	# this happens of detached HEAD
244      	if [[ $BRANCH == HEAD ]]; then
245      		BRANCH="$SEMVER"

make: *** [Makefile:231: pack] Error 1
Error: Process completed with exit code 2.
```

File: .github/workflows/mariner2.yml (hunk 2 above)
Comment (nafraf) [reply .github/workflows/mariner2.yml]: OK. I did the same change. Thanks.

File: .github/workflows/mariner2.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l -eo pipefail {0}
+    container:
+      image: mcr.microsoft.com/cbl-mariner/base/core:2.0
+    steps:
+      - name: Install prereqs
+        run: |
+          tdnf install --noplugins --skipsignature -y ca-certificates
```
Comment (alonre24) [inline .github/workflows/mariner2.yml:21]: Consider haveing these in one line

File: .install/mariner2.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+pip install -q --upgrade setuptools
+pip install -q --upgrade pip
+pip install -q -r tests/pytest/requirements.txt
+
+# These packages are needed to build the package
+pip install -q addict toml jinja2 ramp-packer
+
+# List installed packages
+pip list
```
Comment (alonre24) [inline .install/mariner2.sh:14]: Is this required?

**Review by alonre24 — APPROVED** [review]
